core/src/main/scala/nl/vroste/zio/kinesis/client/Util.scala

```diff
@@ -5,57 +5,6 @@ import zio.stream.ZStream
 
 object Util {
   implicit class ZStreamExtensions[-R, +E, +O](val stream: ZStream[R, E, O]) extends AnyVal {
-    // ZStream's groupBy using distributedWithDynamic is not performant enough, maybe because
```

Comment:
ZIO switched to a much better implementation a while ago. This should no longer be necessary

Reply:
I believe I tried the ZIO implementation at some point after they changed it and still noticed lower performance. Could you run the benchmark/test case to check?

```
ENABLE_AWS=1 sbt
testOnly nl.vroste.zio.kinesis.client.ProducerTest -- -t "efficiently"
```

Reply:
Ran the benchmarks. Seems like a small improvement by using zio builtin groupby.

Old Impl:
interval=299 s, total records published=7890000, throughput (records)=26380.636813192323 records/s, throughput (bytes)=1636555 bytes/s, success rate=100,00 %, failed attempts=0, shard prediction errors=0, mean latency=10409 ms, 95% latency=35391 ms, min latency=114 ms, 2nd attempts=0, max attempts=1, mean payload size=213647 bytes, mean record size=6019 bytes, nr PutRecords calls=2291, mean nr PutRecords calls=7.660080980864843 calls/s

New Impl:
interval=289 s, total records published=8000000, throughput (records)=27656.110098974303 records/s, throughput (bytes)=1719910 bytes/s, success rate=100,00 %, failed attempts=0, shard prediction errors=0, mean latency=10596 ms, 95% latency=46463 ms, min latency=115 ms, 2nd attempts=0, max attempts=1, mean payload size=256980 bytes, mean record size=10563 bytes, nr PutRecords calls=1936, mean nr PutRecords calls=6.692778643951781 calls/s